packages/native/src/types.tsx

```diff
@@ -27,8 +27,16 @@ export type LinkingOptions = {
    * The prefixes are stripped from the URL before parsing them.
    * Usually they are the `scheme` + `host` (e.g. `myapp://chat?user=jane`)
    * Only applicable on Android and iOS.
+   *
+   * @example
+   *  {
+   *    prefixes: [
+   *      "https://example.com", // Exact
+   *      /^[^.s]+.example.com/g // Match with all subdomain and scheme,
```

Comment:
It's just an example, but still:

1. Why do we need `[^.s]+`? Wouldn't `.+` be enough?
1. The `.` needs to be escaped, otherwise it'll match any character
1. I'd rather mention the scheme in the example
1. The `/g` is not necessary since it'll only match once

```suggestion
   *      /^https?:\/\/.+\.example\.com/ // Match with any subdomain and scheme,
```

Reply:
In our case we want to support http too. However, your code is more readable but there is a bug if the url is something like https://other.com/?sub.example.com

Reply:
This regex could solve that`/^(https|http):\/\/[^\/]+.example.com/`

Reply:
So, for this `https://*.example.com` pattern, if we replace * with `[^\/]+` and add `^` to start of prefix we could make our regex internally, after the testing I will push it.

Reply:
> In our case we want to support http too
`https?` matches both `https` and `http`
> there is a bug if the url is something like other.com/?sub.example.com
yeah, `.+` can be changed to `[^/]+` to match anything without `/`. changing to `[^.s]+` doesn't seem to match anything